content/articles/statefull-vs-stateless-architectures/index.md

```diff
@@ -0,0 +1,116 @@
+
+### Network Protocols and their Architectures
+
+Network protocols are divided into two: 
+- Stateful protocols 
+- Stateless protocols. 
+
+### Examples
+  #### Stateless Protocols 
+  
+
+- The Hypertext Transfer Protocol (HTTP): The foundation on which the World Wide Web was built.
+- The User Datagram Protocol: Used in real-time services like video communications 
+- Domain Name System (DNS). 
+- The Simple Mail Transfer Protocol – the foundation of email services.
+- The Post Office Protocol – which is stateless across sessions.
+- The Internet Protocol, which, as the name suggests, is the basis of the internet.
+
+On the other hand, the receiver can keep session state/status info from the previous requests in a stateful network protocol.
+
+#### Statefull Protocols
+- The File Transfer Protocol (FTP).
+
+> The traditional FTP server couples with the client in an interactive session where while in session, a user is provided with a way to verify and set a few variables, all stored server-side, as part of the session state.
+
+- Telnet 
+- The Transmission Control Protocol (TCP)
+
+### API-based Internet Services and their Relation to Stateful and Stateless Architectures
+
+The move to switch from a statefuk designed system to a stteless alternative is linked to the popularity of the diverse ways of building application programming interfaces (APIs) and web apps. At the start of the 21st century, the Simple Object Access Protocol (SOAP) was the prevalent design pattern. The design pattern allowed the creation of stateful architectures, evident from the services and applications that were develooed during the period. 
+
+RESTful services developed in popularity as a result of its stateless design, eventually surpassing SOAP in usage. In the design world, Graph QL is gaining traction. RESTful services, like Graph QL-based services, are stateless thus e nd-users have a lot of influence over what data is sent to and from the servers.
+
+### Differences Between Atateless and Atateful Architectures
+The main difference between the two architectures is how they handle user states and sessions. The differences between the two areas are highlighted in the table below:
+
+**Stateless**|**Stateful**
+:-----:|:-----:
+Rely on clients to store session state information using cookies and local storage.|Rely on servers to store session state information.
+Server design is simple as it does not handle most of the functionality.|Server design is complex as it handles a lot of complexity
+The server handles requests much faster as it does not need to look beyond the single request sent to it as the request contains all the information the server needs to handle it.|The server handles user requests relatively slower compared to stateless architectures servers. The server has to manage the state sessions all on its own, thus, a response gets to the client much slower.
```

Comment:
Break this sentence 'The server handles requests much faster as it does not need to look beyond the single request sent to it as the request contains all the information the server needs to handle it., into two sentences.